src/devices/cpu/xa/xa_ops.cpp

```diff
@@ -1593,7 +1593,35 @@ void xa_cpu::mulu_byte_rd_rs(u8 rd, u8 rs) { fatalerror( "MULU.b %s, %s", m_regn
 //DIVU.b Rd, Rs
 void xa_cpu::divu_byte_rd_rs(u8 rd, u8 rs) { fatalerror( "DIVU.b %s, %s", m_regnames8[rd], m_regnames8[rs]); }
 //MULU.w Rd, Rs
-void xa_cpu::mulu_word_rd_rs(u8 rd, u8 rs) { fatalerror( "MULU.w %s, %s", m_regnames16[rd], m_regnames16[rs]); }
+void xa_cpu::mulu_word_rd_rs(u8 rd, u8 rs)
+{
+	u32 fullval = gr16(rd);
+	u16 rsval = gr16(rs);
+
+	u32 result = fullval * rsval;
+
+	if (result & 0xffff0000)
+		set_v_flag();
+	else
+		clear_v_flag();
+
+	if (result == 0x0000)
+		set_z_flag();
+	else
+		clear_n_flag();
```

Comment:
This looks suspect – set zero, clear negative?

Reply:
looks like a copy+paste from the div, yeah